resources/FocusPoint.php

```diff
@@ -54,40 +54,80 @@ private function focusPointL10n()
      */
     public function getFocusPoint()
     {
-        $attachment = $_POST['attachment'];
+        // Get $_POST['attachment']
+        $attachment = $this->getGlobalPostData('attachment');
 
         // Get the post meta
         $attachment['focusPoint'] = get_post_meta($attachment['id'], 'focus_point', true);
 
-        // Return false
-        if (null === $attachment['id'] || !is_array($attachment['focusPoint'])) {
-            die(json_encode(['success' => false]));
+        $die = json_encode(['success' => false]);
+
+        // Return the focus point if there is one
+        if (null !== $attachment['id'] || is_array($attachment['focusPoint'])) {
+            $die = json_encode([
+                'success'    => true,
+                'focusPoint' => $attachment['focusPoint']
+            ]);
         }
 
-        // Return success
-        die(json_encode([
-            'success'    => true,
-            'focusPoint' => $attachment['focusPoint']
-        ]));
+        // Return the ajax call
+        die($die);
     }
 
     /**
      * Initialize a new crop
      */
     public function initializeCrop()
     {
-        // Check if we've got all the data
-        $attachment = $_POST['attachment'];
+        // Get $_POST['attachment']
+        $attachment = $this->getGlobalPostData('attachment');
+
+        $die = json_encode(['success' => false]);
+
+        // Crop the attachment if there is a focus point
+        if (null !== $attachment['id'] && is_array($attachment['focusPoint'])) {
+            $crop = new CropService();
+            $crop->crop($attachment['id'], $attachment['focusPoint']);
 
-        // Return false
-        if (null === $attachment['focusPoint']) {
-            die(json_encode(['success' => false]));
+            $die = json_encode(['success' => true]);
         }
 
-        $crop = new CropService();
-        $crop->crop($attachment['id'], $attachment['focusPoint']);
+        // Return the ajax call
+        die($die);
+    }
+
+    /**
+     * Sanitize all input that comes in trough $_POST as it might contain harmfull data.
+     *
+     * @param null $postDataKey
+     * @param null $postData
+     * @return null
+     */
+    private function getGlobalPostData($postDataKey = null, $postData = null)
```

Comment:
I thought of that, but for now this is the only class that uses it, and as it requires some edits in the autoload structure etc. I didn't find that neccessary yet.